test/headers.js

```diff
@@ -141,13 +141,3 @@ test('remove null value headers', async t => {
 	const headers = JSON.parse(body);
 	t.false(Reflect.has(headers, 'user-agent'));
 });
-
-test('remove undefined value headers', async t => {
-	const {body} = await got(s.url, {
-		headers: {
-			'user-agent': undefined
-		}
-	});
-	const headers = JSON.parse(body);
-	t.false(Reflect.has(headers, 'user-agent'));
-});
```

Comment:
I wouldn't support this, as it might be confusing with `got.extend({ ... })`, where setting to `undefined` keeps the old value.

Reply:
This test had to be rewritten to use a header key that doesn't have a default value. The behavior is consistent as you wished. :-)